NOnion/TorClient.fs

```diff
@@ -0,0 +1,207 @@
+﻿namespace NOnion
+
+open System
+open System.IO
+open System.Net
+open System.Net.Security
+open System.Net.Sockets
+open System.Reactive.Subjects
+open System.Security.Authentication
+open System.Threading
+
+open NOnion.Cells
+open NOnion.Utility
+
+type TorGuard private (client: TcpClient, sslStream: SslStream) =
+    let shutdownToken = new CancellationTokenSource ()
+
+    let mutable circuitIds: List<uint16> = List.empty
+    let mutable circuitsMap: Map<uint16, TorCircuit> = Map.empty
+    (* Prevents two circuit setup happening at once (to prevent race condition on writing to CircuitIds list) *)
```

Comment:
nit: it's one line only so use the `//` style, not `(*`

Reply:
That file was a duplicated accidently added, it's removed now.

Reply:
you mean removed in the other PR right?